devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -129,20 +129,28 @@
     @Parameter(defaultValue = "false", property = "quarkus.build.skip")
     private boolean skip = false;
 
+    @Parameter(defaultValue = "true", property = "enforce.jar-extension")
+    private boolean enforceJarExtension = true;
+
     public BuildMojo() {
         MojoLogger.logSupplier = this::getLog;
     }
 
     @Override
     public void execute() throws MojoExecutionException {
 
+        if (skip) {
+            getLog().info("Skipping Quarkus build");
+            return;
+        }
         if (project.getPackaging().equals("pom")) {
             getLog().info("Type of the artifact is POM, skipping build goal");
             return;
         }
-        if (skip) {
-            getLog().info("Skipping Quarkus build");
-            return;
+        if (enforceJarExtension && !project.getArtifact().getArtifactHandler().getExtension().equals("jar")) {
```

Comment:
Do we support building the project as a WAR?

Reply:
I don't think so.

Reply:
It seems to me that we might not need a flag but just fail the build if the project isn't packaged as a jar?

Reply:
I also don't think we need the flag.

Reply:
+1 for no flag

Reply:
+1
Just fail hard and make sure the error is clear and visible.

Reply:
BTW I suppose you added the flag out of fear of breaking existing applications?

Fear not, it already doesn't work if try to build it as a war. The only problem currently is that the error is very obscure.

Reply:
Right, I thought I would add a flag to re-enable the previous behavior (which costs nothing) that could be removed in the next version if nobody complained. I didn't specifically think about war packaging/extensions. People do use custom packaging and extensions out there. But, sure, I'll obey :)